src/core/artifact.c

```diff
@@ -420,6 +445,12 @@ mender_artifact_process_data(mender_artifact_ctx_t *ctx, void *input_data, size_
 
                 /* Read manifest file */
                 ret = artifact_read_manifest(ctx);
+
+                /* Early integrity check for version file */
+                if (MENDER_OK != mender_artifact_check_integrity_item(ctx, "version")) {
```

Comment:
I'd add `(ret == MENDER_OK) &&` to the condition because we don't want to do integrity check if the above function fails.